cli/src/alluxio.org/cli/processes/job_masters.go

```diff
@@ -0,0 +1,149 @@
+	cmd.Use = p.Name
+	return cmd
+}
+
+func (p *JobMastersProcess) StopCmd(cmd *cobra.Command) *cobra.Command {
+	cmd.Use = p.Name
+	return cmd
+}
+
+func (p *JobMastersProcess) Start(cmd *env.StartProcessCommand) error {
+	// get list of all masters, stored at mastersList
+	masters, err := getMasters()
+	if err != nil {
+		log.Logger.Fatalf("Cannot get masters, error: %s", err)
+	}
+
+	// get public key for passwordless ssh
+	key, err := getPrivateKey()
+	if err != nil {
+		log.Logger.Fatalf("Cannot get private key, error: %s", err)
+	}
+
+	// generate command
+	cliPath := path.Join(env.Env.EnvVar.GetString(env.ConfAlluxioHome.EnvVar), "bin", "alluxio")
+	arguments := "process start job_master"
+	if cmd.AsyncStart {
+		arguments = arguments + " -a"
+	}
+	if cmd.SkipKillOnStart {
+		arguments = arguments + " -N"
+	}
+	command := cliPath + " " + arguments
+
+	// for each master, create a client and run
+	// TODO: now start master one by one, need to do them in parallel
+	var errors []error
+	for _, master := range masters {
+		conn, err := dialConnection(master, key)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		err = runCommand(master, conn, command)
+		if err != nil {
+			errors = append(errors, err)
+		}
+		err = closeConnection(master, conn)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if len(errors) == 0 {
+		log.Logger.Infof("Run command %s successful on masters: %s", command, masters)
+	} else {
+		log.Logger.Fatalf("Run command %s failed, number of failures: %v", command, len(errors))
+	}
+	return nil
+}
+
+func (p *JobMastersProcess) Stop(cmd *env.StopProcessCommand) error {
+	// get list of all masters, stored at mastersList
+	masters, err := getMasters()
+	if err != nil {
+		log.Logger.Fatalf("Cannot get masters, error: %s", err)
+	}
+
+	// get public key for passwordless ssh
+	key, err := getPrivateKey()
+	if err != nil {
+		log.Logger.Fatalf("Cannot get private key, error: %s", err)
+	}
+
+	// generate command
+	cliPath := path.Join(env.Env.EnvVar.GetString(env.ConfAlluxioHome.EnvVar), "bin", "alluxio")
+	arguments := "process stop job_master"
+	if cmd.SoftKill {
+		arguments = arguments + " -s"
+	}
+	command := cliPath + " " + arguments
+
+	// for each master, create a client and run
+	// TODO: now stop master one by one, need to do them in parallel
+	var errors []error
+	for _, master := range masters {
+		conn, err := dialConnection(master, key)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		err = runCommand(master, conn, command)
+		if err != nil {
+			errors = append(errors, err)
+		}
+		err = closeConnection(master, conn)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if len(errors) == 0 {
+		log.Logger.Infof("Run command %s successful on masters: %s", command, masters)
+	} else {
+		log.Logger.Fatalf("Run command %s failed, number of failures: %v", command, len(errors))
+	}
```

Comment:
can avoid the else block this way
```suggestion
	if len(errors) != 0 {
		return error
	}
	log success
	return nil
```
it is common practice to check for error and handle it, but let the non-error case remain outside the check since it is the expected code path to follow